app/views/shared/menu/_language_selector.html.haml

```diff
@@ -3,6 +3,6 @@
     %i.ofn-i_071-globe
     %span= t 'language_name'
   %ul.dropdown
-    - OpenFoodNetwork::I18nConfig.selectable_locales.each do |l|
+    - OpenFoodNetwork::I18nConfig.selectable_locales.select{|l| l.to_s != I18n.locale.to_s}.each do |l|
```

Comment:
This is a lot of logic in the view. Let's move it to `I18nConfig`. And did you know `reject` which is the opposite of `select`? It's a bit clearer to use `reject` with `==`. But since you are removing an element by value, it can be even simpler:

```suggestion
    - OpenFoodNetwork::I18nConfig.locale_options.each do |l|
```

```diff
diff --git a/lib/open_food_network/i18n_config.rb b/lib/open_food_network/i18n_config.rb
index 8830f9337..83d624f88 100644
--- a/lib/open_food_network/i18n_config.rb
+++ b/lib/open_food_network/i18n_config.rb
@@ -5,6 +5,11 @@ module OpenFoodNetwork
   # Currently, language settings are read from the environment.
   # See: config/application.yml
   class I18nConfig
+    # Users don't need to select the already selected locale.
+    def self.locale_options
+      selectable_locales.delete(I18n.locale.to_s)
+    end
+
     # Locales that can be selected by users.
     def self.selectable_locales
       ENV["AVAILABLE_LOCALES"].andand.split(/[\s,]+/) || []
```

Reply:
Thanks a lot for pointing me out all this things. 
I've committed dd4ca3eac and made some modifications since the `delete` method returns the deleted item in the array and not the array itself.